hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheReplicationOperation.java

```diff
@@ -155,7 +151,11 @@ protected void writeInternal(ObjectDataOutput out)
         int confSize = configs.size();
         out.writeInt(confSize);
         for (CacheConfig config : configs) {
-            out.writeObject(config);
+            if (!classesAlwaysAvailable) {
```

Comment:
As discussed with @lprimak, when `TenantControlFactory#isClassesAlwaysAvailable` is `false`, this operation's serialized form will change to use `PreJoinCacheConfig`. This will break patch-level compatibility with released 3.12.[0-8] versions in case Hazelcast is configured with a `TenantControlFactory`.